src/dune_engine/fs_cache.ml

```diff
@@ -4,26 +4,26 @@ open Import
 
 (* CR-someday amokhov: Implement garbage collection. *)
 
-type 'a t =
+type ('a, 'path) t =
   { name : string (* For debugging *)
-  ; sample : Path.t -> 'a
-  ; cache : 'a Path.Table.t
+  ; sample : 'path -> 'a
+  ; cache : ('path, 'a) Table.t
   ; equal : 'a -> 'a -> bool (* Used to implement cutoff *)
-  ; update_hook : Path.t -> unit (* Run this hook before updating an entry. *)
+  ; update_hook : 'path -> unit (* Run this hook before updating an entry. *)
   }
 
 let create ?(update_hook = fun _path -> ()) name ~sample ~equal =
-  { name; sample; equal; cache = Path.Table.create 128; update_hook }
+  { name; sample; equal; cache = Table.create (module Path) 128; update_hook }
```

Comment:
My understanding is that `create` is still hardcoded to use `Path.t`, so the current PR isn't yet a generalisation.

What other types do you intend to use here? I suppose we might want to switch to `Path.Outside_build_dir.t` at some point. Will we still need `Path.t` after that?.

Reply:
Incidentally, how do `Table.t` and `Path.Table.t` compare in terms of performance? I suppose `Table.t` is a wrapper around pretty much the same type but does that wrapper come with any overheads?

Reply:
> My understanding is that create is still hardcoded to use Path.t, so the current PR isn't yet a generalisation.

Yes, that comes in subsequent PR's.

> What other types do you intend to use here? I suppose we might want to switch to Path.Outside_build_dir.t at some point. Will we still need Path.t after that?

 We won't need `Path.t` after so I can stick to a single type variable and hard code `Path.Outside_build_dir.t`.

> Incidentally, how do Table.t and Path.Table.t compare in terms of performance? I suppose Table.t is a wrapper around pretty much the same type but does that wrapper come with any overheads?

There's a small overhead b/c `Table.t` doesn't allow inlining any of the operations. But in practice, it doesn't really matter to us as we use `Table.t` in many other places that are just as performance sensitive without much problem.

Reply:
> We won't need `Path.t` after so I can stick to a single type variable and hard code `Path.Outside_build_dir.t`.

Let's do that if you don't mind. The extra generality seemed a bit confusing to me.

Reply:
Okay, done.